rust/parquet/src/column/writer.rs

```diff
@@ -98,57 +98,25 @@ pub fn get_column_writer(
 /// Gets a typed column writer for the specific type `T`, by "up-casting" `col_writer` of
 /// non-generic type to a generic column writer type `ColumnWriterImpl`.
 ///
-/// NOTE: the caller MUST guarantee that the actual enum value for `col_writer` matches
-/// the type `T`. Otherwise, disastrous consequence could happen.
+/// Panics if actual enum value for `col_writer` does not match the type `T`.
 pub fn get_typed_column_writer<T: DataType>(
     col_writer: ColumnWriter,
 ) -> ColumnWriterImpl<T> {
-    match col_writer {
-        ColumnWriter::BoolColumnWriter(r) => unsafe { mem::transmute(r) },
-        ColumnWriter::Int32ColumnWriter(r) => unsafe { mem::transmute(r) },
-        ColumnWriter::Int64ColumnWriter(r) => unsafe { mem::transmute(r) },
-        ColumnWriter::Int96ColumnWriter(r) => unsafe { mem::transmute(r) },
-        ColumnWriter::FloatColumnWriter(r) => unsafe { mem::transmute(r) },
-        ColumnWriter::DoubleColumnWriter(r) => unsafe { mem::transmute(r) },
-        ColumnWriter::ByteArrayColumnWriter(r) => unsafe { mem::transmute(r) },
-        ColumnWriter::FixedLenByteArrayColumnWriter(r) => unsafe { mem::transmute(r) },
-    }
+    T::get_column_writer(col_writer).expect("Column type mismatch")
```

Comment:
Could you improve the error message, it is just too generic? Would be good to have something like `Failed to convert column writer into a typed column writer for <data type> type` or something similar. Thanks!